conda/recipes/libkvikio/recipe.yaml

```
@@ -0,0 +1,173 @@
+        # incrementing version number in the compile line doesn't break the
+        # cache
+        set -x
+        export CFLAGS=$(echo $CFLAGS | sed -E 's@\-fdebug\-prefix\-map[^ ]*@@g')
+        export CXXFLAGS=$(echo $CXXFLAGS | sed -E 's@\-fdebug\-prefix\-map[^ ]*@@g')
+        set +x
+
+        ./build.sh -v -n libkvikio
+      secrets:
+        - AWS_ACCESS_KEY_ID
+        - AWS_SECRET_ACCESS_KEY
+        - AWS_SESSION_TOKEN
+      env:
+        CMAKE_C_COMPILER_LAUNCHER: ${{ env.get("CMAKE_C_COMPILER_LAUNCHER") }}
+        CMAKE_CUDA_COMPILER_LAUNCHER: ${{ env.get("CMAKE_CUDA_COMPILER_LAUNCHER") }}
+        CMAKE_CXX_COMPILER_LAUNCHER: ${{ env.get("CMAKE_CXX_COMPILER_LAUNCHER") }}
+        CMAKE_GENERATOR: ${{ env.get("CMAKE_GENERATOR") }}
+        PARALLEL_LEVEL: ${{ env.get("PARALLEL_LEVEL") }}
+        RAPIDS_ARTIFACTS_DIR: ${{ env.get("RAPIDS_ARTIFACTS_DIR") }}
+        SCCACHE_BUCKET: ${{ env.get("SCCACHE_BUCKET") }}
+        SCCACHE_IDLE_TIMEOUT: ${{ env.get("SCCACHE_IDLE_TIMEOUT") }}
+        SCCACHE_REGION: ${{ env.get("SCCACHE_REGION") }}
+        SCCACHE_S3_USE_SSL: ${{ env.get("SCCACHE_S3_USE_SSL") }}
+        SCCACHE_S3_NO_CREDENTIALS: ${{ env.get("SCCACHE_S3_NO_CREDENTIALS") }}
+        SCCACHE_S3_KEY_PREFIX: libkvikio/${{ env.get("RAPIDS_CONDA_ARCH") }}/cuda${{ cuda_major }}
+  requirements:
+    build:
+      - ${{ compiler("c") }}
+      - ${{ compiler("cxx") }}
+      - ${{ compiler("cuda") }}
+      - cuda-version =${{ cuda_version }}
+      - cmake ${{ cmake_version }}
+      - ninja
+      - ${{ stdlib("c") }}
+    host:
+      - cuda-version =${{ cuda_version }}
+      - libcurl ${{ libcurl_version }}
+      - if: cuda_major == "11"
+        then:
+          - cudatoolkit
+          - if: linux64
+            then:
+              - libcufile =${{ cuda11_libcufile_host_version }}
+              - libcufile-dev =${{ cuda11_libcufile_host_version }}
+        else:
+          - if: linux
+            then: libcufile-dev
+
+outputs:
+  - package:
+      name: libkvikio
+      version: ${{ version }}
+    build:
+      script:
+        content: |
+          cmake --install cpp/build
+      string: cuda${{ cuda_major }}_${{ date_string }}_${{ head_rev }}
+      dynamic_linking:
+        overlinking_behavior: "error"
+      prefix_detection:
+        ignore:
+          - lib/libkvikio.so
+    requirements:
+      build:
+        - cmake ${{ cmake_version }}
+        - ${{ compiler("c") }}
+      host:
+        - cuda-version =${{ cuda_version }}
+        - libcurl ${{ libcurl_version }}
+      run:
+        - ${{ pin_compatible("cuda-version", upper_bound="x", lower_bound="x") }}
+        - if: cuda_major == "11"
+          then:
+            - cudatoolkit
+            - if: linux64
+              then:
+                - libcufile ${{ cuda11_libcufile_run_version }}
+                - libcufile-dev ${{ cuda11_libcufile_run_version }}
+          else:
+            - if: linux
+              then: libcufile-dev
```

```suggestion
            - if: linux64
              then: libcufile
```
We probably need to use `libcufile` here. Historically we used the `-dev` package as a runtime dependency because `libkvikio` was header-only, and thus consumers needed `libcufile-dev` any time that `libkvikio` was used (only in host environments). I worked on this in #656, but that PR fell off my radar.

Also, we should only require `libcufile` on linux64, because it did not exist on ARM until CUDA 12.2. We can't easily write "use `libcufile` on ARM if using CUDA 12.2+" as a requirement, afaik. I also fixed in that in #656. There is [a bug that is observable in this PR's CI](https://github.com/rapidsai/kvikio/actions/runs/14206218562/job/39804983799?pr=678#step:9:253) where ARM builds are getting `libcufile 1.9.1.3` from the `nvidia` channel because that channel isn't correctly constrained to `cuda-version` until CUDA 12.5, and the ARM tests are run on `cuda-version==12.0`. Strict priority would've made this more visible.

---

Happy to try it -- I changed it to `linux` (to match the existing recipe) because of the failures here: https://github.com/rapidsai/kvikio/actions/runs/14204454058/job/39799290962?pr=678

but it could be that also switching to `libcufile` fixes those up